consul/kv_consul.go

```diff
@@ -324,17 +417,37 @@ func (kv *consulKV) Update(
 	if err != nil {
 		return nil, err
 	}
+
 	kvPair.Action = kvdb.KVSet
 	return kvPair, nil
 }
 
 func (kv *consulKV) Enumerate(prefix string) (kvdb.KVPairs, error) {
 	prefix = kv.domain + prefix
 	prefix = stripConsecutiveForwardslash(prefix)
-	pairs, meta, err := kv.client.KV().List(prefix, nil)
+	var pairs api.KVPairs
+	var meta *api.QueryMeta
+	var err error
+	for range myParams.machines {
+		pairs, meta, err = kv.client.KV().List(prefix, nil)
+		// *** check error post loop
```

Comment:
same here. We should check for specific errors before retrying on a different endpoint.